frontend/src/store/store-types.ts

```diff
@@ -30,8 +30,39 @@ export type Task = {
   readonly complete: boolean;
   readonly inFocus: boolean; // Whether the task is in focus
   readonly children: Set<string>;
+  readonly type: 'ONE_TIME' | null; // Null for legacy
```

Comment:
You should not put the type here. Similar to the TypeScript in the wiki, you should extract a common Task type and then have something like
```typescript
type RepeatedTask = CommonTask & {
  readonly type: 'REPEATED';
  // ...
}
type OneTimeTask = CommonTask & {
  readonly type: 'ONE_TIME';
  // ...
}
type LegacyTask = CommonTask;
type Task = RepeatedTask | OneTimeTask | LegacyTask;
```